_sources/functions_in_cpp/Exercises.rst

```diff
@@ -0,0 +1,314 @@
+
+      + Since your code is broken down into many functions, rather than debugging the entire program, you can isolate the function that is causing issues.
+
+.. mchoice:: functions_in_cpp_mcq_6
+    :practice: T
+
+    Suppose you have defined the following function:
+
+    ::
+
+        void fortuneCookie (int a, bool b, char c) {
+            if (c < 'm') {
+                if (a % 2 == 0) {
+                    cout << "An alien of some sort will be appearing to you shortly.";
+                }
+                else {
+                    cout << "The fortune you seek is in another cookie.";
+                }
+            }
+            else if (c < 'r') {
+                if (b) {
+                    cout << "He who laughs at himself never runs out of things to laugh at.";
+                }
+                else {
+                    cout << "You will be hungry again in one hour.";
+                }
+            }
+            else {
+                cout << "Fortune not found? Abort, retry, ignore.";
+            }
+        }
+
+    What will be your fortune if you run the following code?
+
+    ::
+
+        fortuneCookie(22, true, 'b');
+
+    -   ``An alien of some sort will be appearing to you shortly.``
+
+        +   ``'b' < 'm'`` and ``22 % 2 == 0``, so this is the fortune that will print.
+
+    -   ``The fortune you seek is in another cookie.``
+
+        -   ``22 % 2 == 0``, so you enter the ``if`` block, not the else.
+
+    -   ``He who laughs at himself never runs out of things to laugh at.``
+
+        -   ``'b'`` is less than ``'m'``, so you would enter the ``if`` block, not the ``else if``.
+
+    -   ``You will be hungry again in one hour.``
+
+        -   ``'b'`` is less than ``'m'``, so you would enter the ``if`` block, not the ``else if``.
+
+    -   ``Fortune not found? Abort, retry, ignore.``
+
+        -   ``'b'`` is less than ``'m'``, so you would enter the ``if`` block, not the ``else``.
+
+.. mchoice:: functions_in_cpp_mcq_7
+    :practice: T
+
+    Are there any issues with the code below?
+
+    .. code-block:: cpp
+
+      bool isEven (int num) {
+        if (num % 2 == 0) {
+          return true;
+        }
+      }
+
+    - Yes, we have to return either 0 or 1.
+
+      - Returning a 0 or 1 would be returning an ``int``, even though booleans evaluate to 0 or 1.
+
+    - Yes, we cannot pass an ``int`` into a ``bool`` function.
+
+      - The type of variables in the parameter list do not affect the return type.
+
+    - Yes, there is no case for odd numbers.
+
+      + Since we never established an else clause, if the input was an odd number, the function would not return anything despite not being a void function.
+
+    - There are no issues with the code.
+
+      - There is an issue with the code. Can you find it?
+
+.. mchoice:: functions_in_cpp_mcq_8
+    :practice: T
+
+    What is the value of ``x`` after the code is run?
+
+    .. code-block:: cpp
+
+      void double(int num) {
```

Comment:
This function needs to be named something different, because `double` is the name of a variable type in C++.